homeassistant/components/sensor/irish_rail_transport.py

```diff
@@ -0,0 +1,183 @@
+                                          destination, stops_at, name)])
+
+
+class IrishRailTransportSensor(Entity):
+    """Implementation of an irish rail public transport sensor."""
+
+    def __init__(self, data, station, direction, destination, stops_at, name):
+        """Initialize the sensor."""
+        self.data = data
+        self._station = station
+        self._direction = direction
+        self._direction = direction
+        self._stops_at = stops_at
+        self._name = name
+        self._state = None
+        self._times = []
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        if len(self._times) > 0:
+            next_up = "None"
+            if len(self._times) > 1:
+                next_up = self._times[1][ATTR_ORIGIN] + " to "
+                next_up += self._times[1][ATTR_DESTINATION] + " in "
+                next_up += self._times[1][ATTR_DUE_IN]
+
+            return {
+                ATTR_STATION: self._station,
+                ATTR_ORIGIN: self._times[0][ATTR_ORIGIN],
+                ATTR_DESTINATION: self._times[0][ATTR_DESTINATION],
+                ATTR_DUE_IN: self._times[0][ATTR_DUE_IN],
+                ATTR_DUE_AT: self._times[0][ATTR_DUE_AT],
+                ATTR_EXPECT_AT: self._times[0][ATTR_EXPECT_AT],
+                ATTR_DIRECTION: self._times[0][ATTR_DIRECTION],
+                ATTR_STOPS_AT: self._times[0][ATTR_STOPS_AT],
+                ATTR_NEXT_UP: next_up,
+                ATTR_TRAIN_TYPE: self._times[0][ATTR_TRAIN_TYPE]
+            }
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit this state is expressed in."""
+        return 'min'
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return ICON
+
+    def update(self):
+        """Get the latest data and update the states."""
+        self.data.update()
+        self._times = self.data.info
+        if len(self._times) > 0:
+            self._state = self._times[0][ATTR_DUE_IN]
+        else:
+            self._state = None
+
+
+class IrishRailTransportData(object):
+    """The Class for handling the data retrieval."""
+
+    def __init__(self, irish_rail, station, direction, destination, stops_at):
+        """Initialize the data object."""
+        self._ir_api = irish_rail
+        self.station = station
+        self.direction = direction
+        self.destination = destination
+        self.stops_at = stops_at
+        self.info = self._empty_train_data()
+
+    def update(self):
+        """Get the latest data from irishrail."""
+        trains = self._ir_api.get_station_by_name(self.station,
```

Comment:
I guess that there will be an exception if the station is unknown.

Reply:
If station name is unknown the API will return an empty list and the sensor will default to the empty train data